### ISSUE TITLE
[Bug] Android Talkback can focus/readout incorrectly

### ISSUE THREAD

**Issue by patchandthat** [xamarin/Xamarin.Forms#9090]
### Description

There is an issue when using Talkback on Android such that some compositions of layouts and elements will cause a single element on the screen to have its accessibility touch area set to the full size of the page. When this happens, touching other elements will often highlight briefly before the focus snaps back to the element that captures the whole screen.

We have this happening on most pages in our production app, which basically makes accessibility a non-starter for us on Android.  

Issue is not present on iOS. Other platforms untested.

Often removing the AutomationProperties from the elements will fix the issue with the touch areas, but then obviously the elements do not have a helpful readout and we're at the mercy of what Voiceover can infer. 

### Steps to Reproduce

See both the MainPage.xaml for some very minimal reproductions, and also ButtonRepro, which is reasonably close the the issue we see in production.

**MainPage.xaml**

1. Run, and enable Talkback
2. Tap the screen, the switch has its touch area set to the whole screen

Remove the AutomationProperties from the switch, and re-reun.  The bounds are set correctly while voiceover is enabled.

**ButtonRepro.xaml**

1. Run and enable Talkback
2. Tap the button in the lower right
3. The label above the button steals the focus, the touch area for the label is the entire page.

Remove, or set IsVisible=False on the stack layout which is the label's parent. Re-run.  All elements on the page have correct talkback touch boundaries.

### Expected Behavior

All accessibility elements have a voiceover touch area which represents their actual visual size on screen.

### Actual Behavior

One accessibility element will have its touch area cover the whole page, and often compete with other elements for focus, ie. Tap on an element, but the focus is immediately stolen by the element which has the whole-page area.

### Basic Information

- Version with issue: Latest stable (4.4.0.991265)
- Last known good version: N/A - Verified as far back as 2.5.1.527436, have not tried earlier versions than this.
- IDE: Visual Studio 2019 (Mac+Windows)
- Platform Target Frameworks: 
  - iOS:  N/A
  - Android: 9.0, 8.1
  - UWP:  N/A
- Android Support Library Version: 
- Nuget Packages: None
- Affected Devices: Reproduced on devices running Android 10 & 5.1

### Screenshots

Observed:
![Screenshot_20200106-135353](https://user-images.githubusercontent.com/9745915/71824563-c1522a00-3091-11ea-8a59-6ab819ab64ad.png)

Expected:
![Screenshot_20200106-135428](https://user-images.githubusercontent.com/9745915/71824588-d29b3680-3091-11ea-8abe-fc7e410682fd.png)

Also:
![Screenshot_20200106-125107](https://user-images.githubusercontent.com/9745915/71824650-024a3e80-3092-11ea-8a13-bdc5c1da634f.png)

### Reproduction Link

https://github.com/patchandthat/jubilant-potato/

See MainPage.xaml for some bare-minimum reproduction cases.  ButtonRepro.xaml is closer to our use case in production.

**Comment by PureWeen** [comment]
I'm guessing this is happening because the *IsInAccessibleTree* is being applied to the ViewGroup instead of the Switch itself and the Switch doesn't have a fast renderer version

**Comment by patchandthat** [comment]
In the second example with the labels and the button, the second label has IsInAccessibleTree=True missing.  Adding it makes that example function correctly.

**Comment by patchandthat** [comment]
@PureWeen It looks like you're correct.  We were seeing the same behaviour on a lot of buttons in our app as with the switch in the sample project.  After a bunch of digging, I see some of our button renderers are derived from the legacy renderers instead of the fast renderers, and changing that fixes things with respect to accessibility.

As a suggestion, is it worth adding `[Obsolete]` to the legacy renderers to help others avoid falling for this?

**Comment by PureWeen** [comment]
ButtonRepro.Xaml still looks like it's an issue 

But the Switch part of this issue looks to be fixed on 5.0.0 most likely by this
https://github.com/xamarin/Xamarin.Forms/pull/13244

**Comment by PureWeen** [comment]
If you haven't already, I also recommend updating to the latest version Xamarin.Forms or at least to Xamarin.Forms 5.0.0.2083, which has many accessibility improvements. These improvements include an experimental accessibility flag that addresses a lot of improvements in Android accessibility.



To leverage this flag on the latest versions of Xamarin Forms, the following code can be inserted in the App.xaml.cs file, following the InitializeComponent call and before the MainPage is set: Xamarin.Forms.Device.SetFlags(new List<string> { "Accessibility_Experimental"});
